src/assignment/components/AssignmentBlockMeta.tsx

```diff
@@ -0,0 +1,55 @@
+import { AssignmentBlock } from '@viaa/avo2-types/types/assignment';
+import React, { FC, useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Link } from 'react-router-dom';
+
+import { APP_PATH } from '../../constants';
+import { SearchFilter } from '../../search/search.const';
+import { buildLink, formatDate } from '../../shared/helpers';
+
+export type AssignmentBlockMetaProps = {
+	block: AssignmentBlock;
+};
+
+export const AssignmentBlockMeta: FC<AssignmentBlockMetaProps> = ({ block }) => {
+	const [t] = useTranslation();
+
+	return useMemo(() => {
```

Comment:
useMemo is een optimalisatie techniek
React is heel goed in het snel vergelijken van virtual dom, dus je moet die niet overal gebruiken. Enkel als er performance issues zijn of als je weet dat de content zo groot is dat het zeker performance issues gaat opleveren.

Want useMemo heeft ook een nadeel en een developer kost.
Als er een developer een extra meta item toevoegt in deze component en vergeet de dependecy list van useMemo te updaten dan zal die de eerste keer wel renderen, maar changes gaat niet zorgen voor een re-render. Wat een bug kan opleveren die de klant pas merkt bij oplevering.

een goeie video: https://youtu.be/DEPwA3mv_R8?t=887